data/binding/json.go

```diff
@@ -0,0 +1,329 @@
+package binding
+
+import (
+	"errors"
+	"sync"
+
+	"fyne.io/fyne/v2/data/binding"
+
+	jsonvalue "github.com/Andrew-M-C/go.jsonvalue"
+)
+
+// JSONValue supports binding a jsonvalue.V{}
+type JSONValue interface {
+	binding.DataItem
+	sync.Locker
+	RLock()
+	RUnlock()
+
+	Get() (*jsonvalue.V, error)
+	Set(*jsonvalue.V) error
+	GetItemString(firstParam interface{}, params ...interface{}) (binding.String, error)
```

Comment:
Should we support more (almost the complete?) list of standard bound types here? It seems like `Int` is missing at least.
Maybe Rune does not make sense, I don't know JSON very well

Reply:
JSON only has three data type: string, float and boolean which is why I got with those three. Any other type, would technically be a conversion, so I left them out since we have all the conversion in fyne itself.

Reply:
I don't think we have a Float<->Int conversion though, so this is going to be a limitation

Reply:
I think JSON number is either int or float isn't it? I.e. I can store "1" as a valid number.

Reply:
Technically an "Int" in JSON is a float without decimal. I can add the logic for an Int here, but I think we should have more conversion support also in Fyne. If you are ok with that, I will also prepare a PR to add FloatToInt and IntToFloat to fyne data binding.